### ISSUE TITLE
writer resource cannot be reused when created from bytearrayresource

### ISSUE THREAD

**Issue by jesseeichar** [jesseeichar/scala-io#9]
The following code snippet fails at the end.  The second acquireFor was expected to write to the file again but instead returned errors.

```
val lines = io.Source.fromFile(new java.io.File("full_dump.dump")).getLines()
val out = Resource.fromFileString("out.dump")
val res3 = out.writer(Codec.UTF8)
res3.acquireFor { writer => 
    lines foreach { line =>
        writer.write(line.replace("public.","otherschema"))
    }
}
val lines = io.Source.fromFile(new java.io.File("full_dump.dump")).getLines()
res3.acquireFor { writer => 
    lines foreach { line =>
        writer.write(line.replace("public.","otherschema"))
        writer.write("\n")
    }
}
```

**Comment by jesseeichar** [comment]
Only occurs when using REPL

**Comment by jesseeichar** [comment]
fixed as of c0b66c593cb81b8a9f33c7c4984bebae622555f0
